sdk/core/Azure.Core/tests/PageableHelpersTests.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using NUnit.Framework;
+
+namespace Azure.Core.Tests
+{
+    public class PageableHelpersTests
+    {
+        [Test]
+        public void PageableHelpers_CreateAsyncPageableFromDelegates()
```

Comment:
In this test, is created from two delegates. It fails to throw `OperationCanceledException` for the token that is provided by `WithCancellation`